src/InternPool.zig

```diff
@@ -205,12 +238,91 @@ pub fn trackZir(
         .tid = tid,
         .index = list.mutate.len,
     }).wrap(ip);
-    list.appendAssumeCapacity(.{key});
+    list.appendAssumeCapacity(.{maybe_lost_key});
     map.entries[map_index] = .{ .value = index.toOptional(), .hash = hash };
     shard.shared.tracked_inst_map.release(new_map);
     return index;
 }
 
+pub fn rehashTrackedInsts(
+    ip: *InternPool,
+    gpa: Allocator,
+    /// TODO: maybe don't take this? it doesn't actually matter, only one thread is running at this point
+    tid: Zcu.PerThread.Id,
+) Allocator.Error!void {
+    // TODO: this function doesn't handle OOM well. What should it do?
+    //       Indeed, what should anyone do when they run out of memory?
+
+    // We don't lock anything, as this function assumes that no other thread is
+    // accessing `tracked_insts`. This is necessary because we're going to be
+    // iterating the `TrackedInst`s in each `Local`, so we have to know that
+    // none will be added as we work.
+
+    // Figure out how big each shard need to be and store it in its mutate `len`.
+    for (ip.shards) |*shard| shard.mutate.tracked_inst_map.len = 0;
+    for (ip.locals) |*local| {
+        // `getMutableTrackedInsts` is okay only because no other thread is currently active.
+        // We need the `mutate` for the len.
+        for (local.getMutableTrackedInsts(gpa).viewAllowEmpty().items(.@"0")) |tracked_inst| {
+            if (tracked_inst.inst == .lost) continue; // we can ignore this one!
+            const full_hash = Hash.hash(0, std.mem.asBytes(&tracked_inst));
+            const shard = &ip.shards[@intCast(full_hash & (ip.shards.len - 1))];
+            shard.mutate.tracked_inst_map.len += 1;
+        }
+    }
+
+    const Map = Shard.Map(TrackedInst.Index.Optional);
+
+    const arena_state = &ip.getLocal(tid).mutate.arena;
```

Comment:
I'm looking at this arena field in the Local.mutate anonymous struct that is documented like this:

```zig
    /// This state is fully local to the owning thread and does not require any
    /// atomic access.
```

Could you document that arena field? I don't understand what is the lifetime of this arena, and why an arena is being used in the first place. Further, why is this the appropriate place for this map of... what?